lib/lambda_ethereum_consensus/validator/proposer.ex

```diff
@@ -4,14 +4,17 @@ defmodule LambdaEthereumConsensus.Validator.Proposer do
   """
   alias LambdaEthereumConsensus.StateTransition.Accessors
   alias LambdaEthereumConsensus.StateTransition.Misc
+  alias LambdaEthereumConsensus.Utils.BitVector
 
   alias Types.BeaconState
 
-  @spec construct_block(BeaconState.t(), Bls.privkey()) :: {:ok, Types.SignedBeaconBlock.t()}
-  def construct_block(state, privkey) do
+  @spec construct_block(BeaconState.t(), Types.validator_index(), Bls.privkey()) ::
+          {:ok, Types.SignedBeaconBlock.t()}
+  def construct_block(%BeaconState{} = state, proposer_index, privkey) do
+    # NOTE: the state is at the start of the block's slot
     block = %Types.BeaconBlock{
-      slot: 1,
-      proposer_index: 63,
+      slot: state.slot,
```

Comment:
maybe we should add a new argument to the function called `slot` instead of messing with the state?

Reply:
Done